src/osemgrep/cli_login/Test_login_subcommand.ml

```diff
@@ -74,31 +58,28 @@ let with_fake_deployment_response return_value f =
  * to move this file out of cli_login/ because of mutual dependencies.
  *)
 let test_logout_not_logged_in caps : Testo.test =
-  t __FUNCTION__
+  t ~checked_output:Stderr
+    ~normalize:[ Testo.mask_not_substring "You are not logged in" ]
+    __FUNCTION__
     (with_login_test_env (fun () ->
-         with_logs
-           ~f:(fun () -> Logout_subcommand.main caps [| "semgrep-logout" |])
-           ~final:(fun res ->
-             assert (res.logs =~ ".*You are not logged in");
```

Comment:
where do we check that the stderr contains "You are not logged in" ?
I see a ~normalize: [Test.mak_not_substring, 
but it's not clear why those 2 functions used together assert that the string is in the stderr.

Reply:
```ocaml
let test_logout_not_logged_in caps : Testo.test =
  t ~checked_output:Stderr
    ~normalize:[ Testo.mask_not_substring "You are not logged in" ]
    __FUNCTION__
    (with_login_test_env (fun () ->
         let exit_code = Logout_subcommand.main caps [| "semgrep-logout" |] in
         assert (exit_code =*= Exit_code.ok)))
```
`~normalize:[ Testo.mask_not_substring "You are not logged in" ]` means the whole output is masked except for `You are not logged in`.

Reply:
I find the testing code a bit more difficult to read than the old code with the simple assert string is in the output.